### ISSUE TITLE
[HITNet

### ISSUE THREAD

**Issue by serizba** [google-research/google-research#613]

**Comment by meteorshowers** [comment]
I meet the same problem.
The error of pb model on sceneflow dataset is higher than the result reported in the hitnet paper.
@serizba Do you meet the same thing?

**Comment by meteorshowers** [comment]
@vtankovich

**Comment by vtankovich** [comment]
> I meet the same problem.
> The error of pb model on sceneflow dataset is higher than the result reported in the hitnet paper.
> @serizba Do you meet the same thing?

Have you ran the evaluate_flyingthings.sh? against TEST set of flyingthings? What numbers have you got at the end?

**Comment by vtankovich** [comment]
> Hi,
> 
> Thanks for publishing this amazing work.
> I have downloaded the FlyingThings model using the provided script. When I feed the model with FlyingThings data I obtain correct and sharp disparity predictions. Nevertheless, if I feed it KITTI data, I obtain incredibly wrong outputs, not even close to the expected predictions.
> 
> Is there any reason your model does not work with images from a different dataset? Do I have to apply some sort of preprocessing before feeding the data to the network?
> 
> Thanks in advance,
> Sergio

The flyingthings model is trained on synthetic only, and is not expected to work on kitti or other real datasets. Currently eth3d and middlebury models are available and are the same ones that were used to generate data for benchmark submission. The middlebury model is pre-trained on flyingthings, as described in paper.

**Comment by serizba** [comment]
Thanks very much for your reply @vtankovich,

I was interested in the generalization performance as reported in the Additional Evaluations B.2 Generalization. I would like to know how I can obtain similar values to those reported in Table 5. When I tried to run KITTI images with the model trained on SceneFlow, results are very far from the ones of the table (I get EPE around 40). Am I missing something?

Thanks
Sergio

**Comment by vtankovich** [comment]
The generalization experiment was done with data augmentation turned on, similar to other pre-training setups in the table. The flyingthings model does not have data augmentations enabled.

**Comment by AvAkanksh** [comment]
@vtankovich  Sir your paper was really good, is there any source code for your model, even I want to reproduce the same kind of results. but I am not able to code the architecture. It would be of great help if you could share your code for training the model. 


Thank you

**Comment by fafancier** [comment]
Hi, @vtankovich  thanks for your amazing work.  Recently, I'm working on the HITNet for stereo matching in **real scene**. So, I am  discovering the **generalization** of it.


The predicted disparity of plant and some other test data are excellent.  
![2021-08-20 13-34-21 的屏幕截图](https://user-images.githubusercontent.com/39083471/130425689-7e286746-b36b-4756-a6ee-5b625bc6f6cb.png)

However, I found that  the result of the pre_trained middlebury model (three models of different disparity are all tested) on the **test set is not so good as paper shows**.  as the following figure shows, left is from Figure 7 in the paper,  where right is the  test result of the model. 

![2021-08-23 17-16-15 的屏幕截图](https://user-images.githubusercontent.com/39083471/130424232-67d67127-2840-454f-b4df-e2546e6bb06c.png)


I also found that the published models not well adapted to illumination change and hue change like this.  

![2021-08-20 13-46-52 的屏幕截图](https://user-images.githubusercontent.com/39083471/130425873-74278038-dadc-48a4-b84a-03d1dc7f1b32.png)

**I'm curious that is the published model not well trained or not the best model? Are the models published not corresponding to the models produce the paper results**. 

Looking forward to your reply.

Thank you!

**Comment by vtankovich** [comment]
@fafancier 
Thanks for your interest in the work. It seems that the model behaves significantly different now (even the plant is not as good as it should be) , compared to Dec 2020. Can you try running it with older versions of the libraries while I'm investigating this?

**Comment by fafancier** [comment]
@vtankovich  thanks for your immediate reply.

you mean even the plant case is also not as good as previous result, and it might due to the different libraries.
  
oh, I forget to mention that I use **tensorflow==2.4.0**, which is not exactly correspond to **tensorflow>=1.11.0 in requirements.txt**. I will give it a try for **tensorflow>=1.11.0** setup.

**Comment by vtankovich** [comment]
The preprocess is part of the .pb file and script. 
I've regenerated the .pb files using current codebase and it behaves closer to the paper. The training set became a bit worse (bad2.0: 6.12 ->6.34), and most of the difference is in PianoL lamp, so hopefully the old checkpoint was loaded correctly and the difference is due to runtime changes. 
I also verified that flying things and eth3d models are not affected, so this may be related to image normalization, which contains division ops.

Summary: middlebury_dXXX.pb now contain models generated with current codebase. Older versions from Dec 2020 are available as middlebury_dXXX_v1.pb.

**Comment by fafancier** [comment]
@vtankovich Thank you for your effort! I will take a try for the new model.
